bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeper.java

```diff
@@ -741,7 +741,8 @@ ZooKeeper getZkHandle() {
         return ((ZKMetadataClientDriver) metadataDriver).getZk();
     }
 
-    protected ClientConfiguration getConf() {
+    @VisibleForTesting
+    public ClientConfiguration getConf() {
```

Comment:
This is not good.
Unfortunately this object is mutable.
You can you Whitebox in tests

Reply:
> This is not good.

forget "VisibleForTesting" annotation, it is very typical to have `getConf` api in a client to retrieve the configuration used by a given client. which I don't think make `getConf` is considered "not good".

I am adding "VisibleForTesting" because this only exists in old bookkeeper api. Ideally the new api should provide `getConf` as well.

Reply:
If I remember correctly sometimes we are picking current values from the configuration.
Maybe it is better to return a copy. We already have the 'features' feature which enable explicitly dynamic reconfiguration.
If you think this is really useful to add it now I am fine. In fact actually the configuration object is already mutable and we do not copy it at instantiation time of the Bookkeeper object.

We can drop the annotation and let it 'public'. We can also add it in the new API.
The only thing I would like is to make it clear in docs that the result must not be changed, and so it would be better to return a copy or an immutable object

Reply:
If I remember correctly sometimes we are picking current values from the configuration.
Maybe it is better to return a copy. We already have the 'features' feature which enable explicitly dynamic reconfiguration.
If you think this is really useful to add it now I am fine. In fact actually the configuration object is already mutable and we do not copy it at instantiation time of the Bookkeeper object.

We can drop the annotation and let it 'public'. We can also add it in the new API.
The only thing I would like is to make it clear in docs that the result must not be changed, and so it would be better to return a copy or an immutable object

Reply:
@eolivelli 

> We can drop the annotation and let it 'public'. We can also add it in the new API.

I see the value of putting `getConf` in the api, I can see it is useful in multiple places, especially when you are using a client in a service component. But I don't want to raise the discussion of exposing this in api, because that should be done in a separate discussion. so I put `@VisibleForTesting` for this PR, which it is the purpose of `@VisibleForTesting`. If you feel really strong about `getConf` or want to get into the whole discussion of `Immutability` of configuration (which doesn't belong this change), I am fine reverting my test code to use whitebox.